cartography/rules/runners.py

```diff
@@ -0,0 +1,264 @@
+
+                if formatted_items:
+                    print(f"      {idx + 1}. {', '.join(formatted_items)}")
+
+            if finding_count > 3:
+                print(
+                    f"      ... and {finding_count - 3} more (use --output json to see all)"
+                )
+        else:
+            print(f"  \033[36m{'Results:':<12}\033[0m No items found")
+
+    # Create and return fact result
+    return FactResult(
+        fact_id=fact.id,
+        fact_name=fact.name,
+        fact_description=fact.description,
+        fact_provider=fact.provider.value,
+        requirement_id=requirement.id,
+        finding_count=finding_count,
+        findings=findings if output_format == "json" else findings[:10],
+        requirement_url=requirement.requirement_url,
+    )
+
+
+def _run_single_framework(
+    framework_name: str,
+    driver: GraphDatabase.driver,
+    database: str,
+    output_format: str,
+    neo4j_uri: str,
+) -> FrameworkResult:
+    """Execute a single framework and return results."""
+    framework = FRAMEWORKS[framework_name]
+
+    # Count total facts for display
+    total_facts = sum(len(req.facts) for req in framework.requirements)
+
+    if output_format == "text":
+        print(f"Executing {framework.name} framework")
+        print(f"Requirements: {len(framework.requirements)}")
+        print(f"Total facts: {total_facts}")
+
+    # Execute facts and collect results
+    total_findings = 0
+    fact_results = []
+    fact_counter = 0
+
+    for requirement in framework.requirements:
+        for fact in requirement.facts:
+            fact_counter += 1
+            fact_result = _run_fact(
+                fact,
+                requirement,
+                framework,
+                driver,
+                database,
+                fact_counter,
+                total_facts,
+                output_format,
+                neo4j_uri,
+            )
+            fact_results.append(fact_result)
+            total_findings += fact_result.finding_count
+
+    # Create and return framework result
+    return FrameworkResult(
+        framework_id=framework.id,
+        framework_name=framework.name,
+        framework_version=framework.version,
+        results=fact_results,
+        total_requirements=len(framework.requirements),
+        total_facts=total_facts,
+        total_findings=total_findings,
+    )
+
+
+def _format_and_output_results(
+    all_results: list[FrameworkResult],
+    framework_names: list[str],
+    output_format: str,
+    total_requirements: int,
+    total_facts: int,
+    total_findings: int,
+):
+    """Format and output the results of framework execution."""
+    if output_format == "json":
+        if len(framework_names) == 1:
+            print(json.dumps(asdict(all_results[0]), indent=2))
+        else:
+            combined_output = [asdict(result) for result in all_results]
+            print(json.dumps(combined_output, indent=2))
```

Comment:
```suggestion
        combined_output = [asdict(result) for result in all_results]
        print(json.dumps(combined_output, indent=2))
```
If someone is using the json output it's likely they're using it for automation. Keeping things array makes the contract more stable:

```
  # With array (consistent):
  cartography-runrules mitre-attack --output json | jq '.[0].total_findings'
  cartography-runrules all --output json | jq '.[].total_findings | add'

  # With mixed format (inconsistent):
  cartography-runrules mitre-attack --output json | jq '.total_findings'     # object
  cartography-runrules all --output json | jq '.[].total_findings | add'     # array
```